contributors/design-proposals/auth/rawproc.md

```diff
@@ -0,0 +1,73 @@
+# RawProc Option
```

Comment:
Other than maybe spending a few minutes trying to come up with a better name, I think this is a reasonable small change.  Allowing containers to nest unprivileged containers is a good thing for other use cases, like isolating child processes.

Reply:
SELinux would protect against most attacks against /proc and more specifically other kernel filesystems like /proc/sys /sys/fs/* ...  But I would be careful making this the norm.

Reply:
@rhatdan and does being non-root protect us at all?

Reply:
Yes, most parts of the kernel file systems require root (CAP_SYS_ADMIN) to do any manipulation of the content.

Reply:
@jessfraz as @rhatdan  mentioned earlier this approach would not work on  a hardened machine with SELinux enabled for docker, and I came up with a strange idea which would still allow to build containers as non-root:
1. Run containerd as non-root on the host (still trying to fix the issues) (so all the issues with masked proc would go away)
2. Run a socat container that would forward the containerd socket to the buider image
3. Img/buildah or whatever tool uses this socket to build images

This is solve many issues, 
1. the containerd process is running as non-root on the host and uses runc to create/build images
2. Do not want to worry/put up hacks for the proc fix
3. SELinux would not be a limiting/blocking factor here

Reply:
Having your build tool run as non root so that it can run inside of an orchestrator like Kubernetes launch builders in non privileged user namespaces, preferably a different UserNS for each builder.  In this case your container runtime like CRI-O or containerd would still be running as root and launching the build containers in different UserNS.

With the tool Buildah, we want to support this method, but we also want to have buildah running as its own Container Runtime, where each buildah container could create the build container in a different UserNS.

Reply:
I agree with @rhatdan the true value comes from running as different unprivileged users not just one

Reply:
I don't know about selinux specifically but apparmor yes

Reply:
I will try on CoreOS w selinux

Reply:
What would a better one-line title be? "Option to disable read only /proc for unprivileged non-root containers"